### DISCUSSION TITLE
scoring

### DISCUSSION THREAD

**Discussion by pgg** [vladmandic/human#246] [Q&A] [answered]
for FaceResult what is the difference between:

score, boxScore and faceScore? more specifically, when detecting a face which is used to determine the quality of the detection? Especially if used for later similarity calculations. The use case here is enrolling a face via detect, make sure it is a fairly good match (score or boxScore or faceScore above required threshold?). Then later using the enrolled embedding to determine similarity . . .

**Comment by vladmandic** [comment, marked as answer]
`boxScore` is score of face detection in overall input.  
`faceScore` is score of face keypoint detection in cropped face and can be missing if `facemesh` model is disabled.  
`score` is simply `faceScore` if its available and if not, then `boxScore`.  

now, you could enroll face in similarity calc without ever running through `facemesh`,  
but running through `facemesh` does have two advantages:  
- it performs slightly more precise face cropping  
- it can optionally perform rotation corrections (if `rotation` is enabled in config)  

neither are not required, but they do help to determine better face descriptors which are used for similarity checks.

**Reply by pgg** [reply]
ok, thanks!

**Comment by plmok61** [comment]
Hi @vladmandic,
I am trying to better understand faceScore and how it should be used. Can this score seen as a confidence that there is a real face in the image? For example if we have above 0.5 for faceScore, we can be pretty confident that we are seeing a real face or am I misunderstanding this value?

**Reply by vladmandic** [reply]
faceScore is confidence of detection of facial keypoints, nothing to do with real or not.
